sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableClientTestBase.java

```diff
@@ -52,12 +53,12 @@ private TableClientBuilder configureTestClientBuilder(TableClientBuilder tableCl
         } else {
             tableClientBuilder.httpClient(buildAssertingClient(DEFAULT_HTTP_CLIENT));
 
-            if (!interceptorManager.isLiveMode()) {
+            if (interceptorManager.isRecordMode()) {
                 recordPolicy = interceptorManager.getRecordPolicy();
                 tableClientBuilder.addPolicy(recordPolicy);
             }
         }
-
+        TestUtils.addTestProxyTestSanitizersAndMatchers(interceptorManager);
```

Comment:
does this not throw?
Ideally matchers are only to be added in playback mode.

Reply:
Nope, I did it like this because currently I implemented matchers and sanitizers within the same function. If you look at the function in TestUtils it only adds matchers in playback.